src/main.rs

```diff
@@ -1,44 +1,80 @@
-use clap::{App, Arg};
+use clap::{crate_authors, crate_version, App, AppSettings, Arg};
+use clap_generate::generators::*;
+use clap_generate::{generate, Generator};
 use std::env;
+use std::io;
 
 mod cli;
 mod git;
 
-fn main() {
-    let matches = App::new("git local-ignore")
+// cli interface
+fn cli_gl() -> App<'static> {
+    App::new("git-local-ignore")
         .version(env!("CARGO_PKG_VERSION"))
         .author("Vyacheslav P. <vyacheslav.pukhanov@gmail.com>")
         .about(
             "Locally exclude files from being tracked by Git (without adding them to .gitignore)",
         )
         .arg(
-            Arg::with_name("force")
+            Arg::new("force")
                 .short('f')
                 .long("force")
                 .about("Ignore any additional prompts, assumes 'yes' as the answer"),
         )
         .arg(
-            Arg::with_name("clear")
+            Arg::new("clear")
                 .conflicts_with_all(&["list", "file"])
                 .short('c')
                 .long("clear")
                 .about("Clear all entries from the exclude file"),
         )
         .arg(
-            Arg::with_name("list")
+            Arg::new("list")
                 .conflicts_with("file")
                 .short('l')
                 .long("list")
                 .about("List all entries in the exclude file"),
         )
         .arg(
-            Arg::with_name("file")
-                .required_unless_one(&["list", "clear"])
-                .index(1)
+            Arg::new("file")
+                //.required_unless_present_any(&["list", "clear"])
+                .short('f')
+                .long("file")
                 .multiple(true)
+                //.index(1)
                 .about("Entries to add to the exclude file"),
         )
-        .get_matches();
+        .subcommand(
+            App::new("completion")
+                .version(crate_version!())
+                .author(crate_authors!())
+                .setting(AppSettings::Hidden)
+                .about("AutoCompletion")
+                .arg(
+                    Arg::new("shell")
+                        .short('s')
+                        .long("shell")
+                        .about("Selects shell")
+                        .required(true)
+                        .takes_value(true)
+                        .possible_values(&["bash", "elvish", "fish", "powershell", "zsh"]),
+                )
+                .arg(
+                    Arg::new("manual")
+                        .short('m')
+                        .long("manual")
+                        .about("Display instructions on how to install autocompletions"),
+                ),
```

Comment:
Maybe we should remove this `manual` arg, since it's not implemented yet anyway? I think that `bash` and `zsh` cover most of the use-cases  anyway 😄

Reply:
Added